src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -105,14 +105,17 @@ def subject(self):
         self._backend.openssl_assert(subject != self._backend._ffi.NULL)
         return _decode_x509_name(self._backend, subject)
 
-    @property
-    def signature_hash_algorithm(self):
+    def _get_signature_algorithm_oid(self):
         alg = self._backend._ffi.new("X509_ALGOR **")
         self._backend._lib.X509_get0_signature(
             self._backend._ffi.NULL, alg, self._x509
         )
         self._backend.openssl_assert(alg[0] != self._backend._ffi.NULL)
-        oid = _obj2txt(self._backend, alg[0].algorithm)
+        return _obj2txt(self._backend, alg[0].algorithm)
+
+    @property
+    def signature_hash_algorithm(self):
+        oid = self._get_signature_algorithm_oid()
         try:
             return x509._SIG_OIDS_TO_HASH[oid]
```

Comment:
Is there a reason no to just have this be `x509._SIG_OIDS_TO_HASH[self.signature_algorithm_oid]` and make that dict be keyed by objects instead of strs?